locked-asset/proxy_dex/src/proxy_farm.rs

```diff
@@ -208,6 +220,63 @@ pub trait ProxyFarmModule:
             .into()
     }
 
+    fn handle_farm_penalty_and_get_output_proxy_farming_token(
+        &self,
+        caller: &ManagedAddress,
+        wrapped_farm_tokens: WrappedFarmToken<Self::Api>,
+        farming_tokens_amount_from_farm: BigUint,
+    ) -> EsdtTokenPayment {
+        require!(
+            wrapped_farm_tokens.payment.amount >= farming_tokens_amount_from_farm,
+            "Invalid payments received from Farm"
+        );
+
+        if wrapped_farm_tokens.payment.amount == farming_tokens_amount_from_farm {
+            return wrapped_farm_tokens.attributes.proxy_farming_token;
+        }
+
+        let penalty_amount = &wrapped_farm_tokens.payment.amount - &farming_tokens_amount_from_farm;
+        let proxy_farming_token = &wrapped_farm_tokens.attributes.proxy_farming_token;
+        let is_locked_token = self
+            .locked_token_ids()
+            .contains(&proxy_farming_token.token_identifier);
+        if is_locked_token {
+            self.burn_locked_tokens_and_update_energy(
+                &proxy_farming_token.token_identifier,
+                proxy_farming_token.token_nonce,
+                &penalty_amount,
+                caller,
+            );
+
+            let mut remaining_locked_tokens = proxy_farming_token.clone();
```

Comment:
This line is the same as the one from L257. Maybe compute it outside and use this variable, as it will be computed either way.